src/routes/console/project-[project]/messaging/topics/topic-[topic]/subscribers/+page.svelte

```diff
@@ -46,9 +46,14 @@
         const targetIds = Object.keys($targetsById).filter(
             (targetId) => !(targetId in subscribersByTargetId)
         );
-        const promises = targetIds.map((targetId) =>
-            sdk.forProject.messaging.createSubscriber($page.params.topic, ID.unique(), targetId)
-        );
+        const promises = targetIds.map(async (targetId) => {
+            const subscriber = await sdk.forProject.messaging.createSubscriber(
+                $page.params.topic,
+                ID.unique(),
+                targetId
+            );
+            subscribersByTargetId[targetId] = subscriber;
+        });
```

Comment:
this will be blocking for each map iteration, we should be able to use a smarter Promise.all approach 👀

Reply:
Each map iteration shouldn't block because`map()` doesn't `await` each callback. If it did, the following would print the numbers in order, right?

<img width="480" alt="image" src="https://github.com/appwrite/console/assets/1477010/41c3e8dc-f6ff-465a-b737-90aad2f6accc">

Reply:
alternatively, i could write this as:

```javascript
        const promises = targetIds.map((targetId) =>
            sdk.forProject.messaging
                .createSubscriber($page.params.topic, ID.unique(), targetId)
                .then((subscriber) => {
                    subscribersByTargetId[targetId] = subscriber;
                })
        );
```

or

```javascript
        const promises = [];
        for (const targetId of targetIds) {
            promises.push(
                sdk.forProject.messaging
                    .createSubscriber($page.params.topic, ID.unique(), targetId)
                    .then((subscriber) => {
                        subscribersByTargetId[targetId] = subscriber;
                    })
            );
        }
```